src/vs/base/browser/ui/list/listView.ts

```diff
@@ -498,7 +498,7 @@ export class ListView<T> implements IListView<T> {
 		this.scrollableElement.delegateVerticalScrollbarPointerDown(browserEvent);
 	}
 
-	updateElementHeight(index: number, size: number | undefined, anchorIndex: number | null): void {
+	updateElementHeight(index: number, size: number | undefined, anchorIndex: number | null, renderTopOverride?: number | undefined): void {
```

Comment:
I'm not a huge fan of conflating view model operations (`updateElementHeight`) with view properties (`renderTopOverride`). Did you find any other way to solve this?

Reply:
In other words, would there be a better default other than `Math.max(0, this.lastRenderTop + heightDiff)` that would fix this issue and play nicely with all other lists?

Reply:
This new behavior will keep the last line in view, and knowing the amount of padding to be able to do that is pretty specific to the type of items in the list, so it would not be straight forward for the listview to take care of this.

Updating element height seems pretty tightly coupled with rendering at a specific render top in the current implementation, but I could look at other options for adjusting those two pieces separately, (like adding the `renderTopOverride` to `rerender`?)

Reply:
@rebornix suggested that we go the other way, and make users of `updateElementHeight` calculate and pass in a render top override if needed. That will simplify `listView`'s `updateElementHeight` implementation and should allow more explicit control to the users of that function, so I like that idea